comp/core/workloadmeta/collectors/internal/kubeapiserver/utils.go

```diff
@@ -57,26 +58,83 @@ func filterToRegex(filter string) (*regexp.Regexp, error) {
 	return r, nil
 }
 
-func discoverGVRs(discoveryClient discovery.DiscoveryInterface, resources []string) ([]schema.GroupVersionResource, error) {
-	discoveredResources, err := discoverResources(discoveryClient)
+func parseRequestedResource(requestedResource string) (group string, version string, resource string) {
+	parts := strings.Split(requestedResource, "/")
+
+	switch len(parts) {
+	case 1:
+		// format is `{resource}`
+		group = ""
+		version = ""
+		resource = parts[0]
+	case 2:
+		// format is `{group}/{resource}`
+		group = parts[0]
+		version = ""
+		resource = parts[1]
+	case 3:
+		// format is `{group}/{version}/{resource}`
+		group = parts[0]
+		version = parts[1]
+		resource = parts[2]
+	default:
+		// format is not correct
```

Comment:
Shouldn't this return an error insted?

Reply:
It can, but I don't think it is necessary. when we return empty group, resource, and version, the function that discovers and constructs GVRs will handle this as an unaccepted input. 

see [here](https://github.com/DataDog/datadog-agent/blob/bfe0aefbf1d1406d960e58140502d2d705d74d82/comp/core/workloadmeta/collectors/internal/kubeapiserver/utils.go#L126-L128).